lib/AST/LookupVisibleDecls.cpp

```diff
@@ -724,7 +724,7 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
 
     // Does it make sense to substitute types?
     bool shouldSubst = !BaseTy->hasUnboundGenericType() &&
-                       !isa<AnyMetatypeType>(BaseTy.getPointer()) &&
+                       !BaseTy->is<AnyMetatypeType>() &&
                        !BaseTy->isAnyExistentialType() &&
                        !BaseTy->hasTypeVariable() &&
                        VD->getDeclContext()->isTypeContext();
```

Comment:
Yes, that's what it was checking - it's possible to substitute only if it's not metatype etc. Will take a look at 28620 and add it to this PR.